desktop/core/src/desktop/auth/backend.py

```diff
@@ -594,6 +594,14 @@ def authenticate(self, request=None, username=None, password=None, server=None):
       LOG.warning("LDAP was not properly configured: %s", detail)
       return None
 
+    if AUTH.PAM_USE_PWD_MODULE.get():
+      LOG.debug('Setting LDAP username to %s using PAM pwd module for user %s' % (getpwnam(username).pw_name, username))
+      pam_user = getpwnam(username).pw_name
+      try:
+        user = User.objects.get(username__iexact=pam_user)
+      except User.DoesNotExist:
```

Comment:
For the record, do we need this part if `find_or_create_user` creates or find?

Reply:
Based on the testing, yes, it is needed there.  I tried to avoid it but without that the login pretty much fails everytime.